pigeon/client.py

```diff
@@ -62,6 +66,19 @@ def __init__(
         )
         self._logger = logger if logger is not None else self._configure_logging()
 
+        self.pid = os.getpid()
+        self.process_name = psutil.Process(self.pid).name()
+        self.hostname = socket.gethostname().split('.')[0]
+        self.name = f'{self.hostname}_{self.pid}_{self.process_name}'
```

Comment:
It would be nice to include `self._service` in the name.

Reply:
I was interested in some other input here as well.
1. Any parent app running a subprocess (python and bokeh) always show up with a process name of the parent.  So all the apps show up as python for example.
2. To that end, I included all components of the "name" in the header because this allows the user to write their own name.  Perhaps if a user wants to customize the name, a setter appends the PID and hostname?
3.